tests/unit/gc/GarbageCollectorTask.spec.js

```diff
@@ -171,6 +172,42 @@ describe('GarbageCollectorTask', () => {
         });
     });
 
+    it('should retry delete operation if gc failed with retryable error', done => {
+        backbeatClient.batchDeleteResponse = { error: { statusCode: 500, retryable: true }, res: null };
+    
+        const entry = ActionQueueEntry.create('deleteArchivedSourceData')
+              .addContext({
+                  origin: 'lifecycle',
+                  ruleType: 'archive',
+                  bucketName: bucket,
+                  objectKey: key,
+                  versionId: version,
+              })
+              .setAttribute('serviceName', 'lifecycle-transition')
+              .setAttribute('target.oldLocation', 'old-location')
+              .setAttribute('target.newLocation', 'new-location')
+              .setAttribute('target.bucket', bucket)
+              .setAttribute('target.key', version)
+              .setAttribute('target.version', key)
+              .setAttribute('target.accountId', accountId)
+              .setAttribute('target.owner', owner);
+    
+        mdObj.setLocation(loc)
+            .setDataStoreName('old-location')
+            .setAmzStorageClass('old-location')
+            .setTransitionInProgress(true);
+        backbeatMetadataProxyClient.setMdObj(mdObj);
+    
+        const batchDeleteDataSpy = sinon.spy(gcTask, '_batchDeleteData');
+    
+        gcTask.processActionEntry(entry, (err, commitInfo) => {
+            assert(batchDeleteDataSpy.callCount > 1);
```

Comment:
Maybe you can check that callcount is equal to maxRetries ? As the function always returns retryable error here